### ISSUE TITLE
showDatePicker widget cuts off

### ISSUE THREAD

**Issue by pgulegin** [flutter/flutter#29681]
## Steps to Reproduce

I am calling showDatePicker() from my Amazon Fire 7.
This issue does not occur on my other (both smaller and larger resolution) devices.
![Screenshot_2019-03-08-15-40-43](https://user-images.githubusercontent.com/22281912/54703468-1a96a000-4af6-11e9-8b5a-9f08c1cfaeaf.png)
![Screenshot_2019-03-08-15-40-48](https://user-images.githubusercontent.com/22281912/54703483-208c8100-4af6-11e9-8edc-277e55c4e525.png)

```
[✓] Flutter (Channel stable, v1.2.1, on Mac OS X 10.14.3 18D109, locale en-US)
    • Flutter version 1.2.1 at /Users/pgulegin/Library/flutter
    • Framework revision 8661d8aecd (5 weeks ago), 2019-02-14 19:19:53 -0800
    • Engine revision 3757390fa4
    • Dart version 2.1.2 (build 2.1.2-dev.0.0 0a7dcf17eb)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/pgulegin/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = /Users/pgulegin/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/pgulegin/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    • ios-deploy 1.9.4
    • CocoaPods version 1.5.2

[✓] Android Studio (version 3.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 33.3.1
    • Dart plugin version 182.5215
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[✓] IntelliJ IDEA Community Edition (version 2018.1.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 24.2.2
    • Dart plugin version 181.4892.1

[✓] VS Code (version 1.32.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.24.0

[✓] Connected device (2 available)
    • KFAUWI           • G0W0MA0772250277                   • android-arm • Android 5.1.1 (API 22)
    • iPhone             • 00008020-001409043A90003A • ios                 • iOS 12.1.4

• No issues found!
```

**Comment by KirolousNashaat** [comment]
same as this :
https://github.com/flutter/flutter/issues/28778 
it may not be about the language

**Comment by darrenaustin** [comment]
This should be fixed with #31514.
